drivers/overlay/encryption.go

```diff
@@ -224,7 +225,9 @@ func programSA(localIP, remoteIP net.IP, spi *spi, k *key, dir int, add bool) (f
 	if add {
 		action = "Adding"
 		xfrmProgram = ns.NlHandle().XfrmStateAdd
-		crypt = &netlink.XfrmStateAlgo{Name: "cbc(aes)", Key: k.value}
+		salt := make([]byte, 4)
+		binary.LittleEndian.PutUint32(salt, k.tag)
+		crypt = &netlink.XfrmStateAlgo{Name: "rfc4106(gcm(aes))", Key: append(k.value, salt...), ICVLen: 64}
```

Comment:
I found this NIST paper helpful when learning about best practices for implementation of AES GCM:  http://csrc.nist.gov/publications/nistpubs/800-38D/SP-800-38D.pdf

Especially Section 8 ("Uniqueness Requirement on IVs and Keys") and the random bit generator construction described in 8.2.2.